src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -1,47 +1,198 @@
     }
 
     @Override
     public T set(T value, int index) {
-        return null;
+        checkAccessIndex(index);
+        final T initialItem;
+        if (index == EMPTY_LIST) {
+            initialItem = first.item;
+            first.item = value;
+            return initialItem;
+        }
+        if (index == size - OFFSET_ONE) {
+            initialItem = last.item;
+            last.item = value;
+            return initialItem;
+        }
+        Node<T> newNode = new Node<>(null, value, null);
+        Node<T> temp = findNodeByIndex(index);
+
+        temp.next.prev = newNode;
+        temp.prev.next = newNode;
+        newNode.prev = temp.prev;
+        newNode.next = temp.next;
+
+        return temp.item;
     }
 
     @Override
     public T remove(int index) {
-        return null;
+        checkAccessIndex(index);
+        if (index == EMPTY_LIST) {
+            return removeFirst();
+        }
+        if (index == size - OFFSET_ONE) {
+            return removeLast();
+        }
+        Node<T> temp = findNodeByIndex(index);
+        unlink(temp);
+        return temp.item;
     }
 
     @Override
     public boolean remove(T object) {
+        Node<T> temp = first;
+        while (temp != null) {
+            if (object != null && temp.equals(first) && object.equals(first.item)) {
+                removeFirst();
+                return true;
+            }
+            if (object != null && temp.equals(last) && object.equals(last.item)) {
+                removeLast();
+                return true;
+            }
+            if (Objects.equals(temp.item, object)) {
+                unlink(temp);
+                return true;
+            }
+            temp = temp.next;
+        }
         return false;
     }
 
     @Override
     public int size() {
-        return 0;
+        return size;
     }
 
     @Override
     public boolean isEmpty() {
-        return false;
+        return size == EMPTY_LIST;
+    }
+
+    private void linkLast(T value) {
+        Node<T> l = last;
+        Node<T> newNode = new Node<>(l, value, null);
+        last = newNode;
+        if (l == null) {
+            first = newNode;
+        } else {
+            l.next = newNode;
+        }
+        size++;
+    }
+
+    private Node<T> findNodeByIndex(int index) {
+        Node<T> temp = first;
+        for (int i = 0; i < index; i++) {
+            temp = temp.next;
+        }
+        return temp;
+    }
+
+    private T removeFirst() {
+        final T initialItem = first.item;
+        if (first.next == null) {
+            first = null;
+            last = null;
+            size--;
+            return initialItem;
+        }
+        first = first.next;
+        size--;
+        return initialItem;
+    }
+
+    private T removeLast() {
+        final T initialItem = last.item;
+        last.prev.next = null;
+        last = last.prev;
+        size--;
+        return initialItem;
+    }
+
+    private void unlink(Node<T> temp) {
```

Comment:
Create a separate method `unlink(Node node)`. You already have this method, which is good. Make sure it's used consistently in your `remove` methods to avoid code duplication.